discovery-infra/tests/base_test.py

```diff
@@ -138,6 +138,21 @@ def attach(node, disk_size, bootable=False):
         for modified_node in modified_nodes:
             modified_node.detach_all_test_disks()
 
+    @pytest.fixture()
+    def attach_interface(self):
+        added_networks = []
+
+        def add(node, network_xml):
+            network, interface_mac = node.attach_interface(node.name, network_xml)
+            added_networks.append({"node": node, "network": network, "mac": interface_mac})
+
+        yield add
+        for added_network in added_networks:
+            logging.info(f'Deleting custom networks:{added_networks}')
+            node_obj = added_network.get("node")
+            node_obj.undefine_interface(node_obj.name, added_network.get("mac"))
+            node_obj.destroy_network(added_network.get("network"))
```

Comment:
What happens if more than one node has an interface on the same network?

Reply:
`destroy_network` run a command so i think nothing will happen

Reply:
Can you destroy a network with an interface attached?

Reply:
havnt tried but i am destroying the interface and than the network